### ISSUE TITLE
Batch job backend upgrade to vllm

### ISSUE THREAD

**Issue by JasonLo** [UW-xDD/text2graph_llm#35]
Batch job on CHTC should use vllm for max speed.

**Comment by JasonLo** [comment]
## Backend spec

- flash_attn (fastest vllm backend)

# Benchmarking plan

1. Evaluate the LLM engine speed exclusively, excluding any unrelated pre/post-processing.
2. Use Tokens Per Second (TPS) as the metric.
3. Vary parameters: `batch_size`, `tensor_parallel_size`, `enforce_eager`.

**Comment by JasonLo** [comment]
On an A100, the chosen setting reaches about 500 TPS. I plan to upgrade the pipeline accordingly.

```python
vllm.LLM(
    model="TheBloke/Mixtral-8x7B-Instruct-v0.1-GPTQ",
    dtype="float16",
    tensor_parallel_size=1,
    enforce_eager=True,
)

mini_batch_size = 200
```

**Comment by iross** [comment]
How's this TPS compare to the current architecture?

**Comment by JasonLo** [comment]
Addressed in #39

**Comment by JasonLo** [comment]
Note to self: Review performance across 100 cases:
- Preprocessing: 3s
- LLM processing: 3 mins
- Post-processing: 30s (still have much potential for optimization).
